AsyncDisplayKit/ASDisplayNode.mm

```diff
@@ -703,6 +704,89 @@ - (instancetype)styledWithBlock:(void (^)(ASLayoutElementStyle *style))styleBloc
 
 #pragma mark - Layout
 
+- (void)invalidateSize
+{
+  ASDisplayNodeAssertThreadAffinity(self);
+  
+  __instanceLock__.lock();
+
+  // Mark the node for layout in the next layout pass
+  [self invalidateCalculatedLayout];
+  
+  if (_supernode) {
+    ASDisplayNode *supernode = _supernode;
```

Comment:
maybe put this outside the condition so only one access of a weak ivar.  Since the only case we won't create the local variable is for the root / _supernode = nil, it should be efficient (although not as visually pretty)